### PR TITLE
Add build examples and build server to runner

### PR DESCRIPTION
**Pull request by tiziano88**
It's not quite on par with the other scripts there, but getting there

Add `./scripts/runner` to make it easier to invoke.

# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [ ] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: runner/src/main.rs [hunk 1]
```diff
@@ -45,15 +181,15 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
...
-    if statuses.contains(&StatusResultValue::Error) {
-        std::process::exit(1);
     }
+}
 
-    Ok(())
+fn examples() -> impl Iterator<Item = PathBuf> {
+    let walker = walkdir::WalkDir::new("./examples").into_iter();
+    walker
+        .filter_entry(|e| !is_ignored_entry(e))
```
Comment (ipetr0v) [inline runner/src/main.rs:190]: Does it also ignore `./examples/certs` ?
https://github.com/tiziano88/oak/blob/a8554dbebeeac6bb9e660e0bc20d548c854ccdb9/runner/src/main.rs#L197-L206

File: runner/src/main.rs [hunk 2]
```diff
@@ -23,16 +23,152 @@
...
+                &[
+                    "build",
+                    "--release",
+                    "--target=x86_64-unknown-linux-musl",
+                    "--package=oak_loader",
+                ],
+            ),
+        },
+        v => {
+            let config = match v {
```
Comment (ipetr0v) [inline runner/src/main.rs:142]: It doesn't yet support the `rust` server, right?

File: runner/src/main.rs [hunk 3]
```diff
@@ -23,16 +23,152 @@
...
+fn bazel_build(extra_build_flags: &[&str], targets: &[&str]) -> Box<dyn Runnable> {
+    let mut flags = Vec::new();
+    flags.push("build");
+
+    let b = bazel_build_flags();
+    flags.extend(as_ref(&b));
+
+    flags.extend(extra_build_flags);
+    flags.push("--");
+    flags.extend(targets);
```
Comment (ipetr0v) [inline runner/src/main.rs:123]: Do we need to also support excluded targets?

File: runner/src/main.rs [hunk 4]
```diff
@@ -23,16 +23,152 @@
...
+}
+
+fn build_examples(opt: &BuildExamples) -> Step {
+    match opt.variant.as_str() {
+        "rust" => Step::Multiple {
+            name: "root".to_string(),
+            steps: examples()
+                .filter(is_cargo_toml_file)
+                .filter(|entry| {
+                    (*entry != PathBuf::from("./examples/aggregator/backend/Cargo.toml"))
```
Comment (ipetr0v) [inline runner/src/main.rs:61]: Could you please add a comment, why these targets are excluded?

File: runner/src/main.rs (hunk 2 above)
Comment (tiziano88) [reply runner/src/main.rs:144]: Not yet (soon!)

File: runner/src/main.rs (hunk 3 above)
Comment (tiziano88) [reply runner/src/main.rs:125]: No need for now (also Bazel will be even less relevant going forward)

File: runner/src/main.rs (hunk 4 above)
Comment (tiziano88) [reply runner/src/main.rs:63]: Done.

File: runner/src/main.rs (hunk 1 above)
Comment (tiziano88) [reply runner/src/main.rs:193]: No, but this is used with extra filtering (only including Cargo.toml files), so in the end that does not affect it.

**Pull request merged**